### PR TITLE
feat: support natural language search for k8s resources

### LINKED ISSUE #452: Feat: Support natural language as a search method for kubernetes resources
### What would you like to be added?

- Design a functional solution using natural language as a search method and implement the functionality
- Write user documentation and reference manuals for all implementations
- If this function is implemented by integrating external AI services, it should be automatically turned off when API_KEY is not given

### Why is this needed?

The original intention of the Karpor project is to provide `users with diverse, user-friendly, and intuitive search methods`, so that they can quickly and conveniently locate the Kubernetes resources they need. 

Although the SQL search function we currently offer does not require additional learning as many engineers already have SQL knowledge, it is clear that the most intuitive and lowest learning threshold search method is to use the user's native language - natural language. 

Therefore, it is particularly necessary to integrate the function of natural language search for Kubernetes resources in Karpor.

### PR DESCRIPTION
**Pull request by ruquanzhao**
## What type of PR is this?
/kind feature

## What this PR does / why we need it:

Integrating AI into our product to reduce the consumption of human attention is a significant milestone for us in the upcoming versions. This PR can be a very good start to build anything related to AI.

Purpose:
- Support an AI backend service that other modules can use to access AI language models.
- Add a new search way: natural language search based on large language models.

## Which issue(s) this PR fixes:

Fixes #452

### CODE REVIEW

File: pkg/kubernetes/registry/types.go [hunk 1]
```diff
@@ -40,7 +40,8 @@ type ExtraConfig struct {
 	ElasticSearchPassword  string
 	ReadOnlyMode           bool
 	GithubBadge            bool
-
+  EnableRBAC             bool
```
Comment (elliotxx) [inline pkg/kubernetes/registry/types.go:43]: Need format

Comment (ruquanzhao) [reply pkg/kubernetes/registry/types.go]: Done

File: pkg/core/handler/search/search.go [hunk 2]
```diff
@@ -18,6 +18,8 @@ import (
 	"net/http"
 	"strconv"
 
+	"github.com/KusionStack/karpor/pkg/core/manager/ai"
+
```
Comment (elliotxx) [inline pkg/core/handler/search/search.go:22]: need format

File: ui/src/pages/result/index.tsx [hunk 3]
```diff
@@ -1,36 +1,110 @@
...
+  return (
+    <div>
+      {options.map((option, index) => (
+        <div
+          key={index}
+          style={{ padding: '5px', borderBottom: '1px solid #ccc' }}
+        >
+          <Option value={option.value}>
+            <span>{option.value}</span> -{' '}
+            <span style={{ color: '#999' }}>{option.value || '默认标签'}</span>
```
Comment (elliotxx) [inline ui/src/pages/result/index.tsx:49]: The '默认标签' needs to support i18n

File: pkg/core/handler/search/search.go (hunk 2 above)
Comment (ruquanzhao) [reply pkg/core/handler/search/search.go]: done

**Review by elliotxx — APPROVED** [review]
lgtm

**Pull request merged**